blob/azureblob/azureblob.go

```diff
@@ -101,9 +101,14 @@ const (
 // Options sets options for constructing a *blob.Bucket backed by Azure Block Blob.
 type Options struct {
 	// Credential represents the authorizer for SignedURL.
-	// Required to use SignedURL.
+	// Required to use SignedURL. If you're using MSI for authentication, this will
+	// attempt to be loaded lazily the first time you call SignedURL.
 	Credential azblob.StorageAccountCredential
 
+	// IsMSIEnvironment represents if the blob storage is being authenticated via
```

Comment:
Is this really an Option? IIUC, you compute it automatically for the URLOpener case, why would we require a user to fill it in when using the constructor?

I.e., can't we drop this and use `adal.MSIAvailable` instead? If that call is expensive, cache it on `bucket`, not `Options`.

Reply:
So the problem I was trying to solve here is that just because MSI is available, we don't want to always use it.

MSI can be available, but you might have not used it to create your credentials.  If this is the case, refreshing the credentials fails.  And then since we don't have access to `bucket` when we use `openerFromMSI` I needed to stick the data somewhere that was available from both contexts.

I've now changed this to only set `credentialExpiration` if we are using delegation credentials, and then we only refresh the token if we have MSI available AND the delegation credentials are being used.